designs/capi-resources-custom-namespace.md

```diff
@@ -0,0 +1,63 @@
+# CAPI Resources in Custom Namespace
+
+## Introduction
+
+**Problem:** Customers managing large scale EKS-A clusters (hundreds or thousands of workload clusters for a given management cluster) expect to have difficulty managing all these resources if they are all in the same namespace. Specifically, they expect to encounter two problems:
+
+1. Difficulty troubleshooting issues for a single cluster when there are so many resources to sift through. 
+2. High levels of load on a bootstrap cluster when executing an upgrade, since the clusterctl move operation will take all the CAPI components inside a namespace and instatiate them on the bootstrap cluster. This can result in heavy and unnecessary load on the bootstrap cluster
+
+This document aims to solve that, by allowing customers to instantiate the underlying CAPI/CAPX/etcd resources in a newly defined namespace, rather than all being dumped into `eksa-system`. 
+This change would make it so customers would be able to put these resources virtually in any namespace (except a few denylisted ones, discussed below), despite the fact that they may be considered eks-a internal components.
+
+### Tenets
+
+* ****Secure By Default:**** make everything simple and secure by default, but give options to change those defaults if there is a need
+
+### Goals and Objectives
+
+* As a EKS-A cluster administrator I want to quickly and easily query resources for a given cluster so that I can more easily troubleshooting cluster issues with many workload clusters present
+* As an EKS-A cluster administrator, I want to be able to perform cluster upgrades without having to move all the cluster resources onto the bootstrap cluster
```

Comment:
It sounds like this user story is controversial - according to Joey, we should continue to move workload cluster capi resources to the bootstrap cluster. Thats the safety mechanism — in case the mgmt cluster itself goes into unrecoverable state, at least we have all the capi resources in the bootstrap cluster

Reply:
This relates to another comment I made: I'm not sure how namespaces solves the problem of upgrading a management cluster.

Reply:
See response above - clusterctl move only moves resources in a single namespace, currently hardcoded to eksa-system.